src/Symfony/Component/Routing/Requirement/EnumRequirement.php

```diff
@@ -22,18 +22,22 @@ final class EnumRequirement implements \Stringable
 
     /**
      * @template T of \BackedEnum
+     *
      * @param class-string<T> $enum
-     * @param T ...$cases
+     * @param list<T>         $cases
      */
-    public function __construct(string $enum, \BackedEnum ...$cases)
+    public function __construct(string $enum, array $cases = [])
     {
-        if (!\is_subclass_of($enum, \BackedEnum::class, true)) {
+        if (!is_subclass_of($enum, \BackedEnum::class, true)) {
             throw new InvalidArgumentException(sprintf('"%s" is not a \BackedEnum class.', $enum));
         }
 
         foreach ($cases as $case) {
+            if (!$case instanceof \BackedEnum) {
+                throw new InvalidArgumentException(sprintf('Case "%s" is not a \BackedEnum class.', \get_debug_type($case)));
+            }
             if (!$case instanceof $enum) {
-                throw new InvalidArgumentException(sprintf('"%s::%s" is not a case of "%s".', \get_class($case), $case->name, $enum));
+                throw new InvalidArgumentException(sprintf('"%s::%s" is not a case of "%s".', \get_debug_type($case), $case->name, $enum));
```

Comment:
```suggestion
                throw new InvalidArgumentException(sprintf('"%s::%s" is not a case of "%s".', \get_class($case), $case->name, $enum));
```
`get_class` is safe here.

Reply:
I have learned from @nicolas-grekas IIRC to always use `get_debug_type()` in favor of `get_class()`. It is bettering in dealing with e.g. anonymous classes.

Reply:
I agree with @wouterj: when a decision is non-conditional (using get_debug_type), everything is simpler.

Reply:
Ok sorry :blush: